### PR TITLE
Caching fixes

### PR DESCRIPTION
**Pull request by jonas**

### CODE REVIEW

File: akka-http-caching/src/main/scala/akka/http/caching/javadsl/CachingSettings.scala [hunk 1]
```diff
@@ -35,8 +37,10 @@ abstract class LfuCacheSettings { self: LfuCacheSettingsImpl ⇒
...
-  /** Java API: Create default caching settings from the systems' configuration */
-  def create(system: ActorSystem): CachingSettings =
-    akka.http.caching.scaladsl.CachingSettings(system)
+object CachingSettings extends SettingsCompanion[CachingSettings] {
+  override def create(config: Config): CachingSettings =
+    akka.http.caching.scaladsl.CachingSettings(config)
+  override def create(configOverrides: String): CachingSettings =
+    akka.http.caching.scaladsl.CachingSettings(configOverrides)
+  override def create(system: ActorSystem): CachingSettings = create(system.settings.config)
 }
```
Comment (ktoso) [inline akka-http-caching/src/main/scala/akka/http/caching/javadsl/CachingSettings.scala:48]: nice, thanks; adding the Java API comment would be nice too

File: akka-http-caching/src/main/scala/akka/http/caching/javadsl/CachingSettings.scala [hunk 2]
```diff
@@ -25,7 +25,7 @@ abstract class CachingSettings private[http] () { self: CachingSettingsImpl ⇒
  * Public API but not intended for subclassing
  */
 @DoNotInherit
-abstract class LfuCacheSettings { self: LfuCacheSettingsImpl ⇒
+abstract class LfuCacheSettings private[http] () { self: LfuCacheSettingsImpl ⇒
```
Comment (ktoso) [inline akka-http-caching/src/main/scala/akka/http/caching/javadsl/CachingSettings.scala:30]: good

File: akka-http-caching/src/main/scala/akka/http/caching/scaladsl/CachingSettings.scala [hunk 3]
```diff
@@ -41,11 +41,12 @@ abstract class LfuCacheSettings private[http] () extends javadsl.LfuCacheSetting
...
-  override def withMaxCapacity(newMaxCapacity: Int): LfuCacheSettings = copy(maxCapacity = newMaxCapacity)
-  override def withInitialCapacity(newInitialCapacity: Int): LfuCacheSettings = copy(initialCapacity = newInitialCapacity)
-  override def withTimeToLive(newTimeToLive: Duration): LfuCacheSettings = copy(timeToLive = newTimeToLive)
-  override def withTimeToIdle(newTimeToIdle: Duration): LfuCacheSettings = copy(timeToIdle = newTimeToIdle)
+  override def withMaxCapacity(newMaxCapacity: Int): LfuCacheSettings = self.copy(maxCapacity = newMaxCapacity)
+  override def withInitialCapacity(newInitialCapacity: Int): LfuCacheSettings = self.copy(initialCapacity = newInitialCapacity)
+  override def withTimeToLive(newTimeToLive: Duration): LfuCacheSettings = self.copy(timeToLive = newTimeToLive)
+  override def withTimeToIdle(newTimeToIdle: Duration): LfuCacheSettings = self.copy(timeToIdle = newTimeToIdle)
 }
+
```
Comment (ktoso) [inline akka-http-caching/src/main/scala/akka/http/caching/scaladsl/CachingSettings.scala:52]: hm, sure I guess

File: akka-http-caching/src/main/scala/akka/http/caching/javadsl/CachingSettings.scala (hunk 1 above)
Comment (jonas) [reply akka-http-caching/src/main/scala/akka/http/caching/javadsl/CachingSettings.scala:48]: True, I thought genjavadoc would somehow put them from the overridden methods, but it seems not.

File: akka-http-caching/src/main/scala/akka/http/caching/scaladsl/CachingSettings.scala [hunk 4]
```diff
@@ -54,7 +55,9 @@ private[http] final case class LfuCacheSettingsImpl(
   initialCapacity: Int,
   timeToLive:      Duration,
   timeToIdle:      Duration
-) extends LfuCacheSettings
+) extends LfuCacheSettings {
+  override def productPrefix = "LfuCacheSettings"
+}
```
Comment (ktoso) [inline akka-http-caching/src/main/scala/akka/http/caching/scaladsl/CachingSettings.scala:62]: good :)

File: docs/src/test/scala/docs/http/scaladsl/server/directives/CachingDirectivesExamplesSpec.scala [hunk 5]
```diff
@@ -60,7 +60,7 @@ class CachingDirectivesExamplesSpec extends RoutingSpec with CachingDirectives {
 
     var i = 0
     val route =
-      alwaysCache(routeCache(), simpleKeyer) {
+      alwaysCache(routeCache, simpleKeyer) {
         complete {
           i += 1
           i.toString
```
Comment (ktoso) [inline docs/src/test/scala/docs/http/scaladsl/server/directives/CachingDirectivesExamplesSpec.scala:66]: good, thanks :)

**Review by ktoso — APPROVED** [review]
Thanks a lot :)

**Pull request merged**